packages/react-google-maps-api/src/GoogleMapNext.tsx

```diff
@@ -0,0 +1,287 @@
+  onClick?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user double-clicks on the map. Note that the click event will also fire, right before this one. */
+  onDblClick?: (e: google.maps.MouseEvent) => void
+  /** This event is repeatedly fired while the user drags the map. */
+  onDrag?: () => void
+  /** This event is fired when the user stops dragging the map. */
+  onDragEnd?: () => void
+  /** This event is fired when the user starts dragging the map. */
+  onDragStart?: () => void
+  /** This event is fired when the mapTypeId property changes. */
+  onMapTypeIdChanged?: () => void
+  /** This event is fired whenever the user's mouse moves over the map container. */
+  onMouseMove?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user's mouse exits the map container. */
+  onMouseOut?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user's mouse enters the map container. */
+  onMouseOver?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the DOM contextmenu event is fired on the map container. */
+  onRightClick?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the visible tiles have finished loading. */
+  onTilesLoaded?: () => void
+  /** This event is fired when the viewport bounds have changed. */
+  onBoundsChanged?: () => void
+  /** This event is fired when the map center property changes. */
+  onCenterChanged?: () => void
+  /** This event is fired when the map heading property changes. */
+  onHeadingChanged?: () => void
+  /** This event is fired when the map becomes idle after panning or zooming. */
+  onIdle?: () => void
+  /** This event is fired when the projection has changed. */
+  onProjectionChanged?: () => void
+  /** This event is fired when the map size has changed. */
+  onResize?: () => void
+  /** This event is fired when the map tilt property changes. */
+  onTiltChanged?: () => void
+  /** This event is fired when the map zoom property changes. */
+  onZoomChanged?: () => void
+  /** This callback is called when the component unmounts. It is called with the map instance. */
+  onLoad?: (map: google.maps.Map) => void | Promise<void>
+  /** This callback is called when the map instance has loaded. It is called with the map instance. */
+  onUnmount?: (map: google.maps.Map) => void | Promise<void>
+}
+
+export function GoogleMap(props: GoogleMapProps): JSX.Element {
+  const [map, setMap] = React.useState<google.maps.Map | null>(null)
+
+  const [registeredEvents, setRedirectEvents] = React.useState<google.maps.MapsEventListener[]>([])
+
+  const mapRef = React.useRef<HTMLDivElement>(null)
+
+  React.useEffect(function onMount () {
```

Comment:
Why do we need to separate mount and update to two separate useEffects functions?
This is a bit counter intuitive to the hooks way of things..

I think they could be united to one useEffect that maybe has a condition to handle mount specific logic